card/src/test/java/com/adyen/checkout/card/internal/ui/DefaultCardDelegateTest.kt

```diff
@@ -809,8 +809,8 @@ internal class DefaultCardDelegateTest(
 
                 with(requireNotNull(paymentComponentData.paymentMethod)) {
                     assertEquals(TEST_CARD_NUMBER, encryptedCardNumber)
-                    assertEquals(TEST_EXPIRY_DATE.expiryMonth.toString(), encryptedExpiryMonth)
-                    assertEquals(TEST_EXPIRY_DATE.expiryYear.toString(), encryptedExpiryYear)
+                    assertEquals(ExpiryDate.from(TEST_EXPIRY_DATE).expiryMonth.toString(), encryptedExpiryMonth)
```

Comment:
Extracting the expiry month using `ExpiryDate.from()` makes this assertion test two things, so if it fails it could be either due to component state not being created properly, or `ExpiryDate.from()` function not working correctly. I think it would be easier to understand if it fails only under one condition.  In order to avoid this we can either use `TEST_EXPIRY_DATE.take(2)` and `TEST_EXPIRY_DATE.takeLast(2)`, or maybe create two more constants for year and month and use it like:
```kotlin
private const val TEST_EXPIRY_MONTH = "03"
private const val TEST_EXPIRY_YEAR = "30"
private const val TEST_EXPIRY_DATE = "$TEST_EXPIRY_MONTH/$TEST_EXPIRY_YEAR"
```
What do you think?

Reply:
What about if we use `trimStart` to remove the `0`s?
```kotlin
assertEquals(TEST_EXPIRY_MONTH.trimStart('0'), encryptedExpiryMonth)
assertEquals(TEST_EXPIRY_YEAR.trimStart('0'), encryptedExpiryYear)
```